lib/faker/travel/train_station.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Faker
+  class Travel
+    class TrainStation < Base
+      class << self
+        ##
+        # Produces random Train Station by name and takes optional arguments for region and type
+        #
+        # @param region [String] Train station region: germany, spain, united_kingdom, united_states
+        #
+        # @param type [String] Train station type: metro, railway
+        #
+        # @return [String]
+        #
+        # @examples
+        # Faker::Travel::TrainStation.name(region: 'united_kingdom', type: 'metro') => "Brockley"
+        # Faker::Travel::TrainStation.name(type: 'railway') => "Düsseldorf Hauptbahnhof"
+        # Faker::Travel::TrainStation.name(region: 'spain') => "Santa Eulàlia"
+        #
+        # @faker.version next
+        def name(region: nil, type: nil)
+          regions = translate('faker.train_station').keys
+          types = translate('faker.train_station.united_kingdom').keys
```

Comment:
I think we can just limit this option to `['metro', 'railway']` for now:
```suggestion
          types = ['metro', 'railway']
```